backend/inspirehep/search/api.py

```diff
@@ -118,19 +123,49 @@ class Meta:
         doc_types = "_doc"
         default_filter = DefaultFilter(Q())
 
-    def query_from_iq(self, query_string):
-        """Initialize ES DSL object using INSPIRE query parser.
-        :param query_string: Query string as a user would input in INSPIRE's search box.
-        :type query_string: string
-        :returns: Elasticsearch DSL search class
-        """
+    def query_for_superuser_or_users(self, query_string):
         if not is_superuser_or_cataloger_logged_in():
             user_query = Q(
                 IQ(query_string, self) & Q("term", _collections="Literature")
             )
             return self.query(user_query)
         return self.query(IQ(query_string, self))
 
+    def query_from_iq(self, query_string):
+        """Initialize ES DSL object using INSPIRE query parser.
+
+        :param query_string: Query string as a user would input in INSPIRE's search box.
+        :type query_string: string
+        :returns: Elasticsearch DSL search class
+        """
+        search_query = self.query_for_superuser_or_users(query_string)
+        search_results = search_query.execute()
+
+        if search_results.hits.total["value"] == 0 and request:
+            reference_match = self.match_reference_from_request_query()
```

Comment:
maybe pass the query string to the method,
```suggestion
            query_string = request.values.get("q", "", type=str)
            reference_match = self.match_reference_from_query(query_string)
```
to make this less magical instead of doing it inside?